src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpArgumentProvider.cs

```diff
@@ -107,6 +107,83 @@ public void Method(IFormatProvider provider)
             VisualStudio.Editor.Verify.CurrentLineText("f.ToString(\"format\"$$)", assertCaretPosition: true);
         }
 
+        [WpfFact]
+        public void FullCycle()
+        {
+            SetUpEditor(@"
+using System;
+public class TestClass
+{
+    public void Method()
+    {$$
+    }
+
+    void Test() { }
+    void Test(int x) { }
+    void Test(int x, int y) { }
+}
+");
+
+            VisualStudio.Editor.SendKeys(VirtualKey.Enter);
+            VisualStudio.Editor.SendKeys("Test");
+
+            VisualStudio.Editor.SendKeys(VirtualKey.Tab);
+            VisualStudio.Editor.Verify.CurrentLineText("Test$$", assertCaretPosition: true);
+
+            VisualStudio.Editor.SendKeys(VirtualKey.Tab);
+            VisualStudio.Workspace.WaitForAllAsyncOperations(Helper.HangMitigatingTimeout, FeatureAttribute.SignatureHelp);
+            VisualStudio.Editor.Verify.CurrentLineText("Test($$)", assertCaretPosition: true);
+
+            VisualStudio.Editor.SendKeys(VirtualKey.Down);
+            VisualStudio.Editor.Verify.CurrentLineText("Test(0$$)", assertCaretPosition: true);
+
+            VisualStudio.Editor.SendKeys(VirtualKey.Down);
+            VisualStudio.Editor.Verify.CurrentLineText("Test(0$$, 0)", assertCaretPosition: true);
+
+            VisualStudio.Editor.SendKeys(VirtualKey.Down);
+            VisualStudio.Editor.Verify.CurrentLineText("Test($$)", assertCaretPosition: true);
+
+            VisualStudio.Editor.SendKeys(VirtualKey.Down);
+            VisualStudio.Editor.Verify.CurrentLineText("Test(0$$)", assertCaretPosition: true);
```

Comment:
📝 If `FormatSpan` is working, this line would fail because the test would have transitioned to:

```patch
-Test(0$$)
+Test()0$$
```